index.js

```diff
@@ -89,4 +97,25 @@ module.exports = function (processors, options) {
   }
 
   return stream
+})
+
+
+function withConfigLoader(cb) {
+  return function (plugins, options) {
+    if (typeof plugins === 'undefined') {
+      return cb(postcssLoadConfig)
```

Comment:
> In both examples everything depends on the filename

In case of `{ modules: true }` thats not the case, it's just a custom context prop to enable CSS Modules in the cssm task with the same `postcss.config.js` used also in the css task. It's just an example for a dynamic config.

> This is such a complicated API with configuration in all the different places

Task and `postcss.config.js`. Communicating via `ctx`

> I may as well just provide a callback for (gulpFile) => { plugins, options } 

`ctx = gulpfile(task => {...custom}) + gulpplugin(file => {ext, dir}) + postcssrc(process => {env})`

`(ctx) => {...options, plugins }`